docs/apps/lnw/es2k/es2k-linux-networking-lag.md

```diff
@@ -0,0 +1,336 @@
+- Both physical ports P0 and P1 should be B2B connected to the Link Partner device and LAG should be configured on both the devices with the associated ports as LAG members using ip link command.
+
+System under test will have above topology running the networking recipe. Link Partner can have the networking recipe or legacy OvS or kernel VxLAN. Note the [Limitations](#limitations) section before setting up the topology.
+
+## Create P4 artifacts and start Infrap4d process
+
+- Use Linux networking p4 program present in the directory `/opt/p4/p4sde/share/mev_reference_p4_files/linux_networking` for this scenario.
+- See [Running Infrap4d on Intel IPU E2100](/guides/es2k/running-infrap4d) for compiling `P4 artifacts`, `bringing up ACC` and running `infrap4d` on ACC.
+
+## Creating the topology
+
+The p4rt-ctl and ovs-vsctl utilities can be found in $P4CP_INSTALL/bin.
+
+### Set the forwarding pipeline
+
+Once the application is started, set the forwarding pipeline config using
+P4Runtime Client `p4rt-ctl` set-pipe command
+
+```bash
+$P4CP_INSTALL/bin/p4rt-ctl set-pipe br0 $OUTPUT_DIR/linux_networking.pb.bin \
+    $OUTPUT_DIR/linux_networking.p4info.txt
+```
+
+Note: Assuming `linux_networking.pb.bin` and `linux_networking.p4info.txt`
+along with other P4 artifacts are created as per the steps mentioned in previous section.
+
+### Configure VSI Group and add a netdev
+
+Use one of the IDPF netdevs on ACC to receive all control packets from overlay
+VM's by assigning to a VSI group. VSI group 3 is dedicated for this configuration,
+execute below devmem commands on IMC.
+
+```bash
+# SEM_DIRECT_MAP_PGEN_CTRL: LSB 11-bit is for vsi which need to map into vsig
+devmem 0x20292002a0 64 0x8000050000000008
+
+# SEM_DIRECT_MAP_PGEN_DATA_VSI_GROUP : This will set vsi
+# (set in SEM_DIRECT_MAP_PGEN_CTRL register LSB) into VSIG-3
+devmem 0x2029200388 64 0x3
+
+# SEM_DIRECT_MAP_PGEN_CTRL: LSB 11-bit is for vsi which need to map into vsig
+devmem 0x20292002a0 64 0xA000050000000008
+```
+
+Note: Here VSI 8 has been used for receiving all control packets and added
+to VSI group 3.  This refers to HOST netdev VSIG 3 as per the topology
+diagram.  Modify this VSI based on your configuration.
+
+### Create Overlay network
+
+Option 1: Create VF's on HOST and spawn VM's on top of those VF's.
+Example to create 4 VF's:  echo 4 > /sys/devices/pci0000:ae/0000:ae:00.0/0000:af:00.0/sriov_numvfs
+
+```bash
+# VM1 configuration
+telnet <VM1 IP> <VM1 port>
+ip addr add 99.0.0.1/24 dev <Netdev connected to VF1>
+ifconfig <Netdev connected to VF> up
+
+# VM2 configuration
+telnet <VM2 IP> <VM2 port>
+ip addr add 99.0.0.2/24 dev <Netdev connected to VF2>
+ifconfig <Netdev connected to VF> up
+```
+
+Option 2: If we are unable to spawn VM's on top of the VF's, for this use case we can also leverage kernel network namespaces.
```

Comment:
```suggestion
Option 2: Using kernel network namespaces
```
has a much more simpler and direct point

Reply:
Addressed.